inlang/source-code/paraglide/paraglide-js/src/vite-plugin/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { PluginOption } from "vite"
+import { exec } from "node:child_process"
+import { readFile } from "node:fs/promises"
+import type { ProjectSettings } from "@inlang/sdk"
+import { resolve } from "node:path"
+
+let cachedSettings: ProjectSettings | undefined = undefined
+
+export const paraglideJsVitePlugin = (config: {
+	settingsPath: string
+	namespace: string
+	silent?: boolean
+	timeout?: number
+	onInit?: boolean
+}): PluginOption => {
+	const options = {
+		silent: false,
+		timeout: 500,
+		onInit: true,
+		...config,
+	}
+
+	let throttled = false
+
+	const execute = () => {
+		exec(
```

Comment:
is there a way to call this directly via the api instead of having to fork out?

Reply:
yeah sure! you can execute compile command by yourself but then you don't have a watcher on your messages.json